File: Bella/Bella.Services/Database/Dying.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Bella.Services.Database
{
    public class Dying
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(7)]
        public string? HexCode { get; set; }
        
        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: Bella/UI/bella_desktop/lib/model/dying.dart
import 'package:json_annotation/json_annotation.dart';

part 'dying.g.dart';

@JsonSerializable()
class Dying {
  final int id;
  final String name;
  final String? hexCode;
  final bool isActive;
  final DateTime createdAt;

  Dying({
    this.id = 0,
    this.name = '',
    this.hexCode,
    this.isActive = true,
    required this.createdAt,
  });

  factory Dying.fromJson(Map<String, dynamic> json) =>
      _$DyingFromJson(json);
  Map<String, dynamic> toJson() => _$DyingToJson(this);
}

